google-cloud-storage/lib/google/cloud/storage/file.rb

```diff
@@ -778,14 +786,19 @@ def public_url protocol: :https
         #
         def signed_url method: nil, expires: nil, content_type: nil,
```

Comment:
Since the additional query parameters don't affect the signed URL's signature, we think its better to add a single optional argument `query` that will apply any additional query parameters:

```ruby
def signed_url method: nil, expires: nil, content_type: nil,
                content_md5: nil, headers: nil, issuer: nil,
                client_email: nil, signing_key: nil, private_key: nil,
                query: nil
```

This is similar to how the `headers` argument is handled.